Framework/Algorithms/test/CrossCorrelateTest.h

```diff
@@ -145,18 +183,75 @@ class CrossCorrelateTest : public CxxTest::TestSuite {
     runAlgorithmThrows(alg);
   }
 
-  void testXMinEqualsXMax() {
-    // this throws because XMin should be > XMax
+  void testValidateInputsXMinEqualsXMax() {
+    // Input validation returns a message because XMin should be < XMax
     CrossCorrelate alg;
-    setupAlgorithm(alg, 2.0, 2.0);
-    runAlgorithmThrows(alg);
+    alg.initialize();
+    alg.setProperty("WorkspaceIndexMin", 0);
+    alg.setProperty("WorkspaceIndexMax", 1);
+    alg.setProperty("XMin", 2.0);
+    alg.setProperty("XMax", 2.0);
+    auto errorMap = alg.validateInputs();
+    TS_ASSERT_EQUALS(errorMap.size(), 2);
+    TS_ASSERT_EQUALS(errorMap.begin()->first, "XMax");
+    TS_ASSERT_EQUALS(errorMap.rbegin()->first, "XMin");
   }
 
-  void testXMinGreaterThanXMax() {
-    // this throws because XMin should be < XMax
+  void testValidateInputsXMinGreaterThanXMax() {
+    // Input validation returns a message because XMin should be < XMax
     CrossCorrelate alg;
-    setupAlgorithm(alg, 3.0, 2.0);
-    runAlgorithmThrows(alg);
+    alg.initialize();
+    alg.setProperty("WorkspaceIndexMin", 0);
+    alg.setProperty("WorkspaceIndexMax", 1);
+    alg.setProperty("XMin", 3.0);
+    alg.setProperty("XMax", 2.0);
+    auto errorMap = alg.validateInputs();
+    TS_ASSERT_EQUALS(errorMap.size(), 2);
+    TS_ASSERT_EQUALS(errorMap.begin()->first, "XMax");
+    TS_ASSERT_EQUALS(errorMap.rbegin()->first, "XMin");
+  }
+
+  void testValidateInputsWSIndexMinEqualsWSIndexMax() {
+    // Input validation returns a message because WorkspaceIndexMin should be < WorkspaceIndexMax
+    CrossCorrelate alg;
+    alg.initialize();
+    alg.setProperty("WorkspaceIndexMin", 1);
+    alg.setProperty("WorkspaceIndexMax", 1);
+    alg.setProperty("XMin", 2.0);
+    alg.setProperty("XMax", 3.0);
+    auto errorMap = alg.validateInputs();
+    TS_ASSERT_EQUALS(errorMap.size(), 2);
+    TS_ASSERT_EQUALS(errorMap.begin()->first, "WorkspaceIndexMax");
```

Comment:
These checks require the order is preserved. `std::map:count("WorkspaceIndexMax") == 1` might be more robust.

Reply:
Thanks, good idea :)